package-res/resources/web/pentaho/type/_type.js

```diff
@@ -414,6 +418,27 @@ define([
 
       _sourceId: null,
 
+      /**
+       * Gets the source module identifier of this type.
+       *
+       * The source identifier is the module id of the _actual_ AMD/RequireJS module
+       * that provides the type and may be different from the [identifier]{@link pentaho.type.Type#id}
+       * when an AMD package or custom mapping is configured for the module.
+       *
+       * The source identifier is used to resolve module ids relative to the source module,
```

Comment:
The source identifier is used to resolve module identifiers relative to the source module,